gobblin-api/src/main/java/gobblin/recordaccess/RecordAccessor.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package gobblin.recordaccess;
+
+import gobblin.annotation.Alpha;
+
+
+/**
+ * Interface for an object that can introspect and manipulate a record. Ideal use case
+ * is for converter logic; converters can specify the operation to do while the
+ * RecordAccessor actually carries it out.
+ *
+ * In general, nested records should be accessible with a '.' separating record name:
+ *  eg given
+ *  "foo": {
+ *    "bar: 1
+ *  }
+ *
+ *  "foo.bar" should refer to integer 1.
+ *
+ * This interface will likely grow over time with more accessors/setters as well as
+ * schema manipulation (for example: rename field or delete field operations).
+ */
+@Alpha
+public interface RecordAccessor {
+  /*
+   * Getters should return null if the field does not exist; may return
+   * ClassCastException if the underlying types do not match. Getters should not
+   * try to do any type coercion -- for example, getAsInt for a value that is the string "1"
```

Comment:
My thought here was to avoid confusion about the underlying schema -- with type coercion getAsInt("stringFieldThatContainsAnInt") could succeed sometimes but fail sometimes depending on record value. Also means that we don't have to worry about type coercion on the set side -- set("stringFieldThatContainsAnInt", 1) may end up failing because Int is the wrong underlying type. What do you think?